docs/msi_with_credential_design.md

```diff
@@ -0,0 +1,233 @@
+
+if (-not $certThumbprint) {
+    Write-Error "❌ Certificate thumbprint is empty. Exiting."
+    exit
+}
+
+Write-Output "🔹 Certificate Thumbprint: $certThumbprint"
+
+# Extract Base64-encoded certificate chain (x5c)
+$x5c = [System.Convert]::ToBase64String($cert.RawData)
+
+$jwk = @{
+    kty = "RSA"
+    use = "sig"
+    alg = "RS256"
+    kid = $cert.Thumbprint
+    x5c = @($x5c)
+} | ConvertTo-Json -Depth 10 -Compress
+
+$body = "{""cnf"": {""jwk"": $jwk}}"
+
+Write-Output "🔹 JSON Payload: $body"
+
+# Requesting MSI credential (Step 1)
+$headers = @{
+    "Metadata" = "true"
+    "X-ms-Client-Request-id" = [guid]::NewGuid().ToString()
+}
+
+$imdsResponse = Invoke-WebRequest -Uri "http://169.254.169.254/metadata/identity/credential?cred-api-version=1.0" `
+    -Method POST `
+    -Headers $headers `
+    -Body $body `
+    -UseBasicParsing
+
+$jsonContent = $imdsResponse.Content | ConvertFrom-Json
+
+$regionalEndpoint = $jsonContent.regional_token_url + "/" + $jsonContent.tenant_id + "/oauth2/v2.0/token"
+$clientId = $jsonContent.client_id
+$credential = $jsonContent.credential
+
+Write-Output "✅ Using Regional Endpoint: $regionalEndpoint"
+
+$resource = "https://management.azure.com"
+$tokenRequestBody = "grant_type=client_credentials&scope=$resource/.default&client_id=$clientId&client_assertion=$credential&client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer"
+
+$tokenHeaders = @{
+    "Content-Type" = "application/x-www-form-urlencoded"
+    "Accept" = "application/json"
+}
+
+try {
+    $tokenResponse = Invoke-WebRequest -Uri $regionalEndpoint `
+        -Method POST `
+        -Headers $tokenHeaders `
+        -Body $tokenRequestBody `
+        -Certificate $cert `
+        -UseBasicParsing
+
+    $tokenJson = $tokenResponse.Content | ConvertFrom-Json
+    Write-Output "🔑 Access Token: $($tokenJson.access_token)"
+} catch {
+    Write-Error "❌ Failed to retrieve access token. Error: $_"
+}
+```
+
+## Summary of New APIs
+
+| API Name                         | Purpose                                                   |
+|----------------------------------|-----------------------------------------------------------|
+| `WithClientCapabilities()`       | Allows client capabilities                                |
+| `WithClaims()`                   | Allows passing of claims (bypasses cache).                |
+| `GetBindingCertificate()`        | Helper method to get the binding certificate.             |
+| `GetManagedIdentitySourceAsync()`| Helper method to get the managed identity source.         |
+| `WithProofOfPossession()`        | Requests a PoP token instead of a default Bearer token.   |
+
+
+## Related Documents
+
+- **[SLC Design Document](https://microsoft.sharepoint.com/:w:/t/AzureMSI/EURnTEtFXPlDngpYhCUioqUBvbSUWEX7vZjP0nm8bxUsQA?e=Ejok1n&wdLOR=cE6820299-49AF-4D7A-B7F7-F58D65C232B6)**
+- **[MSAL EPIC](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/3027078)**
+
```

Comment:
There needs to be a telemetry chapter. At a minimum we'd want to know if we are dealing with a self-generated cert or one read from the store. Anything else?